[Bug]: ClickHouse v2 storage panics on empty or short TraceIDs/SpanIDs during row conversion

### What happened?

While reviewing the ClickHouse v2 storage implementation, I found a panic in the read path.

In `internal/storage/v2/clickhouse/tracestore/dbmodel/from.go`, the `convertSpan` function uses `hex.DecodeString` to read `sr.TraceID` and `sr.ID` into byte slices. It immediately casts these slices to `pcommon.TraceID` and `pcommon.SpanID` (which are aliases for `[16]byte` and `[8]byte` arrays):

```go
traceId, err := hex.DecodeString(sr.TraceID) // returns []byte
// ...
span.SetTraceID(pcommon.TraceID(traceId)) // Panics if len(traceId) < 16
```

If `sr.TraceID` is empty (length 0) or shorter than 32 hex characters, `hex.DecodeString` returns a slice shorter than 16 bytes with no error. However, casting a short slice to a fixed `[16]byte` array causes a runtime panic in Go.

Because this happens on the read path, a single corrupted row or empty `TraceID` in ClickHouse will completely crash the `jaeger-query` (or all-in-one) process when it attempts to load that trace.

### Steps to reproduce

1. Run Jaeger with the ClickHouse v2 storage backend.
2. Manually insert a single row into the ClickHouse `spans` table with an empty `trace_id` (or one shorter than 32 hex chars).
3. Call the `GetTrace` or `FindTraces` API that attempts to read that row.
4. Observe that the Jaeger process crashes with a slice-to-array out-of-bounds panic.

### Expected behavior

Jaeger should gracefully handle malformed data from the storage layer. It should either drop the corrupted span
with a warning (using `jptrace.AddWarnings`) or safely pad/sanitize the ID, rather than allowing a runtime panic to crash the entire service.

### Relevant log output

```shell
panic: runtime error: cannot convert slice with length 0 to array or pointer to array with length 16
    
goroutine 123 [running]:
github.com/jaegertracing/jaeger/internal/storage/v2/clickhouse/tracestore/dbmodel.convertSpan(...)
/go/src/github.com/jaegertracing/jaeger/internal/storage/v2/clickhouse/tracestore/dbmodel/from.go:76
github.com/jaegertracing/jaeger/internal/storage/v2/clickhouse/tracestore/dbmodel.FromRow(...)
/go/src/github.com/jaegertracing/jaeger/internal/storage/v2/clickhouse/tracestore/dbmodel/from.go:28
```

### Screenshot

_No response_

### Additional context

The fix is straightforward. We just need to check the length of the decoded slice before casting, and either copy
it into a fixed  [16]byte  buffer safely or return an error/warning. 

Happy to open a PR to fix this for `TraceID`, `SpanID`, `ParentSpanID`, and `Link` IDs in `from.go`. 

### Jaeger backend version

v2 (latest)

### SDK

_No response_

### Pipeline

_No response_

### Stogage backend

ClickHouse v2

### Operating system

Linux / macOS

### Deployment model

Local development / Docker

### Deployment configs

```shell

```

--- a/internal/storage/v2/clickhouse/tracestore/dbmodel/from.go
+++ b/internal/storage/v2/clickhouse/tracestore/dbmodel/from.go
@@ -66,25 +66,55 @@ func convertScope(sr *SpanRow, spanForWarnings ptrace.Span) pcommon.Instrumentat
 	return scope
 }
 
+// decodeTraceID decodes a hex string into a pcommon.TraceID, validating that
+// it contains exactly 16 bytes so the conversion cannot panic on corrupted rows.
+func decodeTraceID(s string) (pcommon.TraceID, error) {
+	var id pcommon.TraceID
+	b, err := hex.DecodeString(s)
+	if err != nil {
+		return id, err
+	}
+	if len(b) != len(id) {
+		return id, fmt.Errorf("invalid length %d of decoded trace ID %q, expected %d bytes", len(b), s, len(id))
+	}
+	copy(id[:], b)
+	return id, nil
+}
+
+// decodeSpanID decodes a hex string into a pcommon.SpanID, validating that
+// it contains exactly 8 bytes so the conversion cannot panic on corrupted rows.
+func decodeSpanID(s string) (pcommon.SpanID, error) {
+	var id pcommon.SpanID
+	b, err := hex.DecodeString(s)
+	if err != nil {
+		return id, err
+	}
+	if len(b) != len(id) {
+		return id, fmt.Errorf("invalid length %d of decoded span ID %q, expected %d bytes", len(b), s, len(id))
+	}
+	copy(id[:], b)
+	return id, nil
+}
+
 func convertSpan(sr *SpanRow) (ptrace.Span, error) {
 	span := ptrace.NewSpan()
 	span.SetStartTimestamp(pcommon.NewTimestampFromTime(sr.StartTime))
-	traceId, err := hex.DecodeString(sr.TraceID)
+	traceId, err := decodeTraceID(sr.TraceID)
 	if err != nil {
 		return span, fmt.Errorf("failed to decode trace ID: %w", err)
 	}
-	span.SetTraceID(pcommon.TraceID(traceId))
-	spanId, err := hex.DecodeString(sr.ID)
+	span.SetTraceID(traceId)
+	spanId, err := decodeSpanID(sr.ID)
 	if err != nil {
 		return span, fmt.Errorf("failed to decode span ID: %w", err)
 	}
-	span.SetSpanID(pcommon.SpanID(spanId))
-	parentSpanId, err := hex.DecodeString(sr.ParentSpanID)
-	if err != nil {
-		return span, fmt.Errorf("failed to decode parent span ID: %w", err)
-	}
-	if len(parentSpanId) != 0 {
-		span.SetParentSpanID(pcommon.SpanID(parentSpanId))
+	span.SetSpanID(spanId)
+	if sr.ParentSpanID != "" {
+		parentSpanId, err := decodeSpanID(sr.ParentSpanID)
+		if err != nil {
+			return span, fmt.Errorf("failed to decode parent span ID: %w", err)
+		}
+		span.SetParentSpanID(parentSpanId)
 	}
 	span.TraceState().FromRaw(sr.TraceState)
 	span.SetName(sr.Name)
@@ -108,18 +138,18 @@ func convertSpan(sr *SpanRow) (ptrace.Span, error) {
 
 	for i, l := range sr.LinkTraceIDs {
 		link := span.Links().AppendEmpty()
-		traceID, err := hex.DecodeString(l)
+		traceID, err := decodeTraceID(l)
 		if err != nil {
 			jptrace.AddWarnings(span, fmt.Sprintf("failed to decode link trace ID: %v", err))
 			continue
 		}
-		link.SetTraceID(pcommon.TraceID(traceID))
-		spanID, err := hex.DecodeString(sr.LinkSpanIDs[i])
+		link.SetTraceID(traceID)
+		spanID, err := decodeSpanID(sr.LinkSpanIDs[i])
 		if err != nil {
 			jptrace.AddWarnings(span, fmt.Sprintf("failed to decode link span ID: %v", err))
 			continue
 		}
-		link.SetSpanID(pcommon.SpanID(spanID))
+		link.SetSpanID(spanID)
 		link.TraceState().FromRaw(sr.LinkTraceStates[i])
 
 		putAttributes2D(link.Attributes(), &sr.LinkAttributes, i, span)

--- a/internal/storage/v2/clickhouse/tracestore/dbmodel/from_test.go
+++ b/internal/storage/v2/clickhouse/tracestore/dbmodel/from_test.go
@@ -77,15 +77,117 @@ func TestFromRow_DecodeID(t *testing.T) {
 			},
 			want: "failed to decode link span ID: encoding/hex: invalid byte: U+0078 'x'",
 		},
+		{
+			name: "empty trace id",
+			arg: &SpanRow{
+				TraceID: "",
+			},
+			want: `failed to decode trace ID: invalid length 0 of decoded trace ID "", expected 16 bytes`,
+		},
+		{
+			name: "too short trace id",
+			arg: &SpanRow{
+				TraceID: "0001",
+			},
+			want: `failed to decode trace ID: invalid length 2 of decoded trace ID "0001", expected 16 bytes`,
+		},
+		{
+			name: "too long trace id",
+			arg: &SpanRow{
+				TraceID: "000100010001000100010001000100010001",
+			},
+			want: `failed to decode trace ID: invalid length 18 of decoded trace ID "000100010001000100010001000100010001", expected 16 bytes`,
+		},
+		{
+			name: "empty span id",
+			arg: &SpanRow{
+				TraceID: "00010001000100010001000100010001",
+				ID:      "",
+			},
+			want: `failed to decode span ID: invalid length 0 of decoded span ID "", expected 8 bytes`,
+		},
+		{
+			name: "too short span id",
+			arg: &SpanRow{
+				TraceID: "00010001000100010001000100010001",
+				ID:      "0001",
+			},
+			want: `failed to decode span ID: invalid length 2 of decoded span ID "0001", expected 8 bytes`,
+		},
+		{
+			name: "too short parent span id",
+			arg: &SpanRow{
+				TraceID:      "00010001000100010001000100010001",
+				ID:           "0001000100010001",
+				ParentSpanID: "0001",
+			},
+			want: `failed to decode parent span ID: invalid length 2 of decoded span ID "0001", expected 8 bytes`,
+		},
+		{
+			name: "empty link trace id",
+			arg: &SpanRow{
+				TraceID:      "00010001000100010001000100010001",
+				ID:           "0001000100010001",
+				ParentSpanID: "0001000100010001",
+				LinkTraceIDs: []string{""},
+			},
+			want: `failed to decode link trace ID: invalid length 0 of decoded trace ID "", expected 16 bytes`,
+		},
+		{
+			name: "too short link trace id",
+			arg: &SpanRow{
+				TraceID:      "00010001000100010001000100010001",
+				ID:           "0001000100010001",
+				ParentSpanID: "0001000100010001",
+				LinkTraceIDs: []string{"0001"},
+			},
+			want: `failed to decode link trace ID: invalid length 2 of decoded trace ID "0001", expected 16 bytes`,
+		},
+		{
+			name: "empty link span id",
+			arg: &SpanRow{
+				TraceID:      "00010001000100010001000100010001",
+				ID:           "0001000100010001",
+				ParentSpanID: "0001000100010001",
+				LinkTraceIDs: []string{"00010001000100010001000100010001"},
+				LinkSpanIDs:  []string{""},
+			},
+			want: `failed to decode link span ID: invalid length 0 of decoded span ID "", expected 8 bytes`,
+		},
+		{
+			name: "too short link span id",
+			arg: &SpanRow{
+				TraceID:      "00010001000100010001000100010001",
+				ID:           "0001000100010001",
+				ParentSpanID: "0001000100010001",
+				LinkTraceIDs: []string{"00010001000100010001000100010001"},
+				LinkSpanIDs:  []string{"0001"},
+			},
+			want: `failed to decode link span ID: invalid length 2 of decoded span ID "0001", expected 8 bytes`,
+		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			trace := FromRow(tt.arg)
+			var trace ptrace.Traces
+			require.NotPanics(t, func() {
+				trace = FromRow(tt.arg)
+			})
 			span := trace.ResourceSpans().At(0).ScopeSpans().At(0).Spans().At(0)
 			require.Contains(t, jptrace.GetWarnings(span), tt.want)
 		})
 	}
+}
+
+func TestFromRow_EmptyParentSpanID(t *testing.T) {
+	trace := FromRow(&SpanRow{
+		TraceID:      "00010001000100010001000100010001",
+		ID:           "0001000100010001",
+		ParentSpanID: "",
+	})
+	span := trace.ResourceSpans().At(0).ScopeSpans().At(0).Spans().At(0)
+	require.True(t, span.ParentSpanID().IsEmpty())
+	require.Empty(t, jptrace.GetWarnings(span))
 }
 
 func TestPutAttributes_Warnings(t *testing.T) {

